gems/aws-sdk-core/lib/aws-sdk-core/log/param_filter.rb

```diff
@@ -31,7 +31,7 @@ def filter(value)
       def filter_hash(values)
         filtered = {}
         values.each_pair do |key, value|
-          filtered[key] = @filters.include?(key) ? '[FILTERED]' : filter(value)
+          filtered[key] = @filters.any? { |f| f.to_s.casecmp(key.to_s).zero? } ? '[FILTERED]' : filter(value)
```

Comment:
What happens in the Array case? Would we potentially want to filter those?

Reply:
I'll write a test for the array case to see what happens and adjust according.

Reply:
In the array case, hash/Struct/nested arrays elements are filtered due to the recursive `#filter` call.

If the object is a primitive value (say a `String`), then the value is returned as-is.

I have a basic test case incoming.